truss/remote/baseten/api.py

```diff
@@ -566,3 +567,27 @@ def get_all_secrets(self) -> Any:
 
         secrets_info = resp.json()
         return secrets_info
+
+    def upsert_training_project(self, training_project):
```

Comment:
I personally think the rest API structure should mirror the truss SDK very closely. The server code is already going to have to transform things (i.e. get the actual instance type, worker planes, user/org, etc), but we can keep the truss integration simple (basically a model_dump() here instead of explicit transformation code)